sql/sql_explain.cc

```diff
@@ -351,9 +351,9 @@ int print_explain_row(select_result_sink *result,
                       mem_root);
   
   /* 'possible_keys' */
+  StringBuffer<64> possible_keys_buf;
   if (possible_keys && !possible_keys->is_empty())
   {
-    StringBuffer<64> possible_keys_buf;
```

Comment:
I would suggest compiling all code with -fsanitize=undefined, then running all tests and `grep -i runtime var/*/log/mysqld*err` for UBSAN messages. Note that you would probably want to silence messages for `uint4korr()` and friends.
I recently conducted such an experiment for InnoDB; see commit 7edadde72eb23e9110db8cea810c023104e9d15e and others mentioned in MDEV-12358.